src/EventAccumulator.ts

```diff
@@ -28,10 +29,64 @@ export class EventAccumulator implements IEventAccumulator {
    * @param {CountlyWebSdk} metricsService - instance
    * @param {number} flushInterval - in milliseconds
    */
-  constructor (metricsService: CountlyWebSdk, flushInterval: number = 5 * 60 * 1000) {
+  constructor (metricsService: CountlyWebSdk | CountlyNodeSdk, flushInterval: number = 5 * 60 * 1000) {
     this.metricsService = metricsService
     this.flushInterval = flushInterval
-    this.events = {}
+    this.events = new Map()
+    this.setupUnloadEvent()
+  }
+
+  private setupUnloadEvent (): void {
+    globalThis.addEventListener('beforeunload', () => {
+      this.flushAll()
+    })
+  }
+
+  /**
+   * Pad events with default values
+   *
+   * @param {CountlyEvent} event - event to add defaults to
+   * @returns {CountlyEventData} - event with defaults added
+   */
+  private addEventDefaults (event: CountlyEvent): CountlyEventData {
+    return { ...eventDefaults, ...event }
+  }
+
+  /**
+   * Setup the event accumulator for a key type for the first time.
+   *
+   * @param {CountlyEventData} eventData - event data
+   */
+  private setupEventAccumulator (eventData: CountlyEventData): void {
+    const { key } = eventData
+    this.events.set(key, {
+      eventData,
+      // set start time to now. This will be updated when the event is flushed.
+      startTime: Date.now(),
+      // set a timeout to flush the event after the flush interval.
+      timeout: setTimeout(() => {
+        this.flush(key)
+      }, this.flushInterval)
+    })
+  }
+
+  /**
+   * Digest only the event data from an event.
+   *
+   * @param {CountlyEventData} newEventData - event data
+   */
+  private digestEventData (newEventData: CountlyEventData): void {
```

Comment:
nit: digest is a significantly overloaded term in our space and doesn't really communicate what this function does under most of the common definitions. Could we use `accumulate` (not accurate for new events) or `process` (vague), or even `updateEventData` (somewhat vague)?

A better description in the jsdoc that doesn't use the term "digest" could help a lot here.